### PR TITLE
[QA] Fix the dependency error from langgraph

### PR DESCRIPTION
**Pull request by lingyielia**
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/pyproject.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ dependencies = [
   "pandas",
   "tabulate",
   "openai>=1.0.0",
+  "langgraph-checkpoint<=1.0.12",  # # TODO update to pydantic v2 and remove upper bound(https://github.com/langchain-ai/langgraph/issues/1914)
```
Comment (huong-li-nguyen) [inline vizro-ai/pyproject.toml:20]:
```suggestion
  "langgraph-checkpoint<=1.0.12",  # TODO update to pydantic v2 and remove upper bound(https://github.com/langchain-ai/langgraph/issues/1914)
```

**Review by huong-li-nguyen — APPROVED** [review]

**Comment by maxschulz-COL** [comment]
Let's see the reply to [this](https://github.com/langchain-ai/langgraph/issues/1914#issuecomment-2384967710) first - hopefully soon. It seems to me that they did not obey semantic versioning here. If there is no answer, then let's pin the dependency with a comment to remove once they have fixed it.

**Comment by huong-li-nguyen** [comment]
> this

I would just merge it for now given that we were aiming to release vizro-core yesterday. Would like to have at least 1-2 puffer das of testing before the GHC. We can still change that later in a separate PR, but currently, we need this fix so the main branch is not broken.

File: vizro-ai/pyproject.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ dependencies = [
   "pandas",
   "tabulate",
   "openai>=1.0.0",
+  "langgraph-checkpoint<=1.0.12",  # TODO update to pydantic v2 and remove upper bound(https://github.com/langchain-ai/langgraph/issues/1914)
```
Comment (maxschulz-COL) [inline vizro-ai/pyproject.toml:20]:
```suggestion
  "langgraph-checkpoint<=1.0.12",  # TODO update to pydantic v2 and remove upper bound (https://github.com/langchain-ai/langgraph/issues/1914)
```

**Review by maxschulz-COL — APPROVED** [review]
Happy to merge now that we have the comment in

**Pull request merged**